agent/resources/resources_linux.go

```diff
@@ -55,13 +68,32 @@ func (c *cgroupWrapper) Cleanup(task *api.Task) error {
 	return c.cleanupCgroup(task)
 }
 
-// cgroupInit is used to create the root '/ecs/ cgroup
+func (c *cgroupWrapper) ApplyConfigDependencies(cfg *config.Config) {
+	c.cgroupPath = cfg.CgroupPath
+}
+
+// cgroupInit is used to create the root '/ecs/ cgroup and enable
+// memory.use_hierarchy at the '/ecs/' level
 func (c *cgroupWrapper) cgroupInit() error {
 	if c.control.Exists(config.DefaultTaskCgroupPrefix) {
 		seelog.Debugf("Cgroup at %s already exists, skipping creation", config.DefaultTaskCgroupPrefix)
 		return nil
 	}
-	return c.control.Init()
+	err := c.control.Init()
+
+	if err != nil {
+		return err
+	}
+
+	// echo 1 > memory.use_hierarchy
+	cgroupRoot := c.cgroupPath
+
+	err = c.ioutil.WriteFile(filepath.Join(cgroupRoot, memorySubsystem, config.DefaultTaskCgroupPrefix, memoryUseHierarchy), []byte(strconv.Itoa(1)), os.FileMode(0))
```

Comment:
nit: please split this into multiple lines. Also, please add documentation for all those magic numbers:
* `[]byte(strconv.Itoa(1)`
* `os.FileMode(0)`

Reply:
oops missed this one for splits, and will doc numbers.